src/D2L.CodeStyle.Analyzers/Immutability/ExplicitImmutableAttributeTransitivityAnalyzer.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using D2L.CodeStyle.Analyzers.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.Analyzers.Immutability {
+	// [Immutable] is a pledge (with checking) that values of your type *or any
+	// subtype* are immutable. In other words the effects/checks for
+	// [Immutable] apply transitively to your subtypes. Rather than this
+	// being implicit, we make it explicit by requiring your subtypes to be
+	// annotated with [Immutable]. This is useful documentation and keeps the
+	// implementation for the mutability analysis simple.
+
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	internal sealed class ExplicitImmutableAttributeTransitivityAnalyzer : DiagnosticAnalyzer {
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+			=> ImmutableArray.Create(
+				Diagnostics.MissingTransitiveImmutableAttribute
+			);
+
+		public override void Initialize( AnalysisContext context ) {
+			//context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction( CompilationStart );
+		}
+
+		public void CompilationStart(
+			CompilationStartAnalysisContext context
+		) {
+			var immutableAttribute = context.Compilation.GetTypeByMetadataName(

```

Comment:
FYI: I get really annoyed seeing analyzers compare symbols via strings. This is the proper way to do it. The immutability stuff does string-based stuff a lot.